tpcx_bb/xbb_tools/utils.py

```diff
@@ -198,10 +440,528 @@ def get_query_number():
+        except AssertionError as error:
+            print("Error", error)
+            print("Correctness Assertion False")
+
+    return result_verified
+
+
+#################################
+# Performance Tracking Automation
+#################################
+
+
+def build_benchmark_googlesheet_payload(cli_args):
+    """
+    cli_args : dict
+    """
+    # Don't mutate original dictionary
+    data = cli_args.copy()
+
+    # get the hostname of the machine running this workload
+    data["hostname"] = socket.gethostname()
+
+    QUERY_NUM = int(get_query_number())
+    current_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+    query_time = _get_benchmarked_method_time(
+        filename="benchmarked_main.csv", query_start_time=cli_args.get("start_time")
+    )
+    writing_time = _get_benchmarked_method_time(
+        filename="benchmarked_write_result.csv",
+        query_start_time=cli_args.get("start_time"),
+    )
+    read_graph_creation_time = _get_benchmarked_method_time(
+        filename="benchmarked_read_tables.csv",
+        query_start_time=cli_args.get("start_time"),
+    )
+
+    if data["get_read_time"] and read_graph_creation_time and query_time:
+        ### below contains the computation time
+        compute_read_table_time = _get_benchmarked_method_time(
+            filename="benchmarked_read_tables.csv",
+            field="compute_time_seconds",
+            query_start_time=cli_args.get("start_time"),
+        )
+        # subtracting read calculation time
+        query_time = query_time - compute_read_table_time
+    else:
+        compute_read_table_time = None
+
+    # get library info
+    library_info = generate_library_information()
+    data.update(library_info)
+
+    payload = OrderedDict(
+        {
+            "Query Number": QUERY_NUM,
+            "Protocol": data.get("protocol"),
+            "NVLINK": data.get("nvlink", "NA"),
+            "Infiniband": data.get("infiniband", "NA"),
+            "Query Type": "blazing" if is_blazing_query() else "dask",
+            "File Format": data.get("file_format"),
+            "Time (seconds)": query_time + writing_time
+            if query_time and writing_time
+            else "NA",
+            "Query Time(seconds)": query_time if query_time else "NA",
+            "Writing Results Time": writing_time if writing_time else "NA",
+            # read time
+            "Compute Read + Repartition small table Time(seconds)": compute_read_table_time
+            if compute_read_table_time
+            else "NA",
+            "Graph Creation time(seconds)": read_graph_creation_time
+            if read_graph_creation_time
+            else "NA",
+            "Machine Setup": data.get("hostname"),
+            "Data Location": data.get("data_dir"),
+            "RMM Pool": data.get("pool_mode"),
+            "RMM Pool Size": data.get("pool_size_str"),
+            "RMM Allocator Mode": data.get("rmm_allocator_mode"),
```

Comment:
This all may be redundant as this may only be set when we create cluster which we dont do anymore on a query level .